src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/TypeCast.cs

```diff
@@ -967,13 +912,6 @@ private static unsafe EETypeElementType GetNormalizedIntegralArrayElementType(Me
             return elementType;
         }
 
-        // Would not be inlined, but still need to mark NoInlining so that it doesn't throw off tail calls
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        private static unsafe object ThrowInvalidCastException(MethodTable* pMT)
```

Comment:
ThrowHelper.ThrowInvalidCastException looks different (doesn't return object, is not marked noinlining). Are we deoptimizing codegen or are the underlying RyuJIT issues fixed up?

Reply:
RyuJIT understands and special cases throw helpers that aren't marked with `NoInlining` so this should improve the codegen or keep it the same.

Reply:
I've checked and the codegen is different, the JIT now puts the throw in a cold block at the end of the method instead of a tailcall in the middle. I'd argue it's better since the need for a jump for the exception shouldn't matter much.